// chayns-components/lib/react-chayns-calendar/component/Month.js
import React from 'react';
import PropTypes from 'prop-types';

import MonthTable from './MonthTable';

var Month = function Month(_ref) {
    var title = _ref.title,
        className = _ref.className,
        onDateSelect = _ref.onDateSelect,
        startDate = _ref.startDate,
        endDate = _ref.endDate,
        selected = _ref.selected,
        activated = _ref.activated,
        highlighted = _ref.highlighted,
        activateAll = _ref.activateAll;
    return React.createElement(
        'div',
        { className: 'month__item ' + (className || '') },
        React.createElement(
            'div',
            { className: 'month__title' },
            title || ''
        ),
        React.createElement(MonthTable, {
            onDateSelect: onDateSelect,
            startDate: startDate,
            endDate: endDate,
            selected: selected,
            activated: activated,
            highlighted: highlighted,
            activateAll: activateAll
        })
    );
};

Month.defaultProps = {
    title: '',
    className: '',
    selected: null,
    activated: false,
    highlighted: false,
    activateAll: null
};

export default Month;
