// chayns-components/lib/react-chayns-animations/component/InspectElement/ModalTransition.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { Component } from 'react';
import { Transition } from 'react-transition-group';
import Modal from './Modal';

var ModalTransition = function (_Component) {
    _inherits(ModalTransition, _Component);

    function ModalTransition() {
        var _ref;

        var _temp, _this, _ret;

        _classCallCheck(this, ModalTransition);

        for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
            args[_key] = arguments[_key];
        }

        return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = ModalTransition.__proto__ || Object.getPrototypeOf(ModalTransition)).call.apply(_ref, [this].concat(args))), _this), _this.state = {}, _temp), _possibleConstructorReturn(_this, _ret);
    }

    _createClass(ModalTransition, [{
        key: 'render',
        value: function render() {
            var _this2 = this;

            return React.createElement(
                Transition,
                _extends({
                    timeout: 650,
                    mountOnEnter: true,
                    unmountOnExit: true,
                    appear: true
                }, this.props),
                function (status) {
                    return React.createElement(Modal, _extends({
                        status: status
                    }, _this2.props));
                }
            );
        }
    }]);

    return ModalTransition;
}(Component);

export default ModalTransition;
