// chayns-components/lib/react-chayns-gallery/component/Gallery.js
var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MoreImages from './MoreImages';
import ImageContainer from './ImageContainer';

var Gallery = (_temp = _class = function (_Component) {
    _inherits(Gallery, _Component);

    function Gallery() {
        _classCallCheck(this, Gallery);

        var _this = _possibleConstructorReturn(this, (Gallery.__proto__ || Object.getPrototypeOf(Gallery)).call(this));

        _this.openFirstImage = _this.openGallery.bind(_this, 0);
        _this.openSecondImage = _this.openGallery.bind(_this, 1);
        _this.openThirdImage = _this.openGallery.bind(_this, 2);
        return _this;
    }

    _createClass(Gallery, [{
        key: 'openGallery',
        value: function openGallery(start) {
            var _props = this.props,
                onClick = _props.onClick,
                urls = _props.urls;


            onClick(urls, start);
        }
    }, {
        key: 'render',
        value: function render() {
            var _props2 = this.props,
                urls = _props2.urls,
                onlyIcon = _props2.onlyIcon;

            var count = urls.length;

            return React.createElement(
                'div',
                { className: 'chayns-gallery' },
                React.createElement(
                    'div',
                    { className: 'gallery-grid' },
                    React.createElement(ImageContainer, {
                        className: 'gallery_item',
                        url: urls[0],
                        onClick: this.openFirstImage
                    }),
                    count > 1 && React.createElement(ImageContainer, {
                        className: 'gallery_item',
                        url: urls[1],
                        onClick: this.openSecondImage
                    }),
                    count > 2 && React.createElement(
                        ImageContainer,
                        {
                            className: 'gallery_item',
                            url: urls[2],
                            onClick: this.openThirdImage
                        },
                        count > 3 && React.createElement(MoreImages, {
                            count: count - 3,
                            onlyIcon: onlyIcon
                        })
                    )
                )
            );
        }
    }]);

    return Gallery;
}(Component), _class.defaultProps = {
    onlyIcon: false,
    onClick: chayns.openImage
}, _temp);
export { Gallery as default };
