// chayns-components/lib/react-chayns-selectlist/component/internal/SelectItemInternal.js
var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React from 'react';
import PropTypes from 'prop-types';
import { CSSTransitionGroup } from 'react-transition-group';

var SelectItemInternal = (_temp = _class = function (_React$Component) {
    _inherits(SelectItemInternal, _React$Component);

    function SelectItemInternal(props) {
        _classCallCheck(this, SelectItemInternal);

        var _this = _possibleConstructorReturn(this, (SelectItemInternal.__proto__ || Object.getPrototypeOf(SelectItemInternal)).call(this, props));

        _this._handleChange = function () {
            var _this$props = _this.props,
                onChange = _this$props.onChange,
                id = _this$props.id,
                value = _this$props.value;

            onChange(id, value);
        };

        _this.radioId = _this._getRadioId(props.id);
        return _this;
    }

    _createClass(SelectItemInternal, [{
        key: '_renderChildren',
        value: function _renderChildren() {
            var _props = this.props,
                checked = _props.checked,
                children = _props.children;


            if (checked) {
                return children;
            }

            return null;
        }
    }, {
        key: '_getRadioId',
        value: function _getRadioId(id) {
            return this.props.selectListId + '-' + id;
        }
    }, {
        key: 'render',
        value: function render() {
            var _props2 = this.props,
                id = _props2.id,
                className = _props2.className,
                selectListId = _props2.selectListId,
                disabled = _props2.disabled,
                name = _props2.name,
                checked = _props2.checked;


            return React.createElement(
                'div',
                {
                    key: id,
                    className: className
                },
                React.createElement('input', {
                    name: selectListId,
                    type: 'radio',
                    className: 'radio',
                    id: this.radioId,
                    checked: checked,
                    onChange: this._handleChange,
                    disabled: disabled
                }),
                React.createElement(
                    'label',
                    { htmlFor: this.radioId },
                    name
                ),
                React.createElement(
                    CSSTransitionGroup,
                    {
                        transitionName: 'react-fade',
                        transitionEnterTimeout: 500,
                        transitionLeaveTimeout: 500
                    },
                    this._renderChildren()
                )
            );
        }
    }]);

    return SelectItemInternal;
}(React.Component), _class.defaultProps = {
    id: null,
    className: '',
    checked: false,
    disabled: false,
    children: null,
    name: '',
    value: null
}, _temp);
export { SelectItemInternal as default };
