// chayns-components/lib/react-chayns-image_upload/constants/colors.js
export var GRAY_1 = {
    rgb: [110, 110, 110],
    hex: '#6e6e6e',
    name: 'Grau',
    id: 'GRAY_1'
};

export var RED_1 = {
    rgb: [172, 0, 0],
    hex: '#ac0000',
    name: 'Rot',
    id: 'RED_1'
};

export var GREEN_1 = {
    rgb: [55, 145, 60],
    hex: '#37913c',
    name: 'Hellgrün',
    id: 'GREEN_1'
};

export var ORANGE_1 = {
    rgb: [255, 129, 26],
    hex: '#ff811a',
    name: 'Orange',
    id: 'ORANGE_1'
};

export var PURPLE_1 = {
    rgb: [94, 72, 131],
    hex: '#5e4883',
    name: 'Lila 1',
    id: 'PURPLE_1'
};

export var BEIGE_1 = {
    rgb: [131, 104, 73],
    hex: '#836849',
    name: 'Beige',
    id: 'BEIGE_1'
};

export var BLUE_1 = {
    rgb: [0, 85, 164],
    hex: '#0055a4',
    name: 'Blau 1',
    id: 'BLUE_1'
};

export var BLUE_2 = {
    rgb: [0, 158, 244],
    hex: '#009ee0',
    name: 'Blau 2',
    id: 'BLUE_2'
};

export var GREEN_2 = {
    rgb: [0, 77, 48],
    hex: '#004d30',
    name: 'Dunkelgrün',
    id: 'GREEN_2'
};

export var CYAN_1 = {
    rgb: [51, 115, 128],
    hex: '#337380',
    name: 'Türkis',
    id: 'CYAN_1'
};

export var PURPLE_2 = {
    rgb: [46, 46, 103],
    hex: '#2e2e67',
    name: 'Lila 2',
    id: 'PURPLE_2'
};

export var RED_2 = {
    rgb: [115, 3, 31],
    hex: '#73031f',
    name: 'Weinrot',
    id: 'RED_2'
};

export var PINK_1 = {
    rgb: [209, 31, 135],
    hex: '#d11f87',
    name: 'Pink 1',
    id: 'PINK_1'
};

export var PINK_2 = {
    rgb: [140, 35, 114],
    hex: '#8c2372',
    name: 'Pink 2',
    id: 'PINK_2'
};

export var YELLOW_1 = {
    rgb: [230, 180, 64],
    hex: '#e6b440',
    name: 'Gelb',
    id: 'YELLOW_1'
};

export var BROWN_1 = {
    rgb: [77, 36, 23],
    hex: '#4d2417',
    name: 'Braun',
    id: 'BROWN_1'
};

export var COLORS = [GRAY_1, RED_1, GREEN_1, ORANGE_1, PURPLE_1, BEIGE_1, BLUE_1, BLUE_2, GREEN_2, CYAN_1, PURPLE_2, RED_2, PINK_1, PINK_2, YELLOW_1, BROWN_1];
