// chayns-components/lib/react-chayns-contextmenu/component/ContextMenu.js
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import PropTypes from 'prop-types';

import ContextMenuItem from './ContextMenuItem';

var ContextMenu = function ContextMenu(_ref) {
    var hide = _ref.hide,
        onLayerClick = _ref.onLayerClick,
        x = _ref.x,
        y = _ref.y,
        items = _ref.items;

    var contextMenuClass = hide ? 'context-menu' : 'context-menu context-menu--active';

    return React.createElement(
        'div',
        { className: contextMenuClass },
        React.createElement('div', {
            className: 'context-menu__page-block',
            onClick: !chayns.env.isMobile ? onLayerClick : null,
            onTouchStart: chayns.env.isMobile ? onLayerClick : null
        }),
        React.createElement(
            'div',
            {
                className: 'context-menu__body',
                style: {
                    position: 'absolute',
                    left: x,
                    top: y
                }
            },
            React.createElement(
                'ul',
                { className: 'context-menu__item-list' },
                items.map(function (item, index) {
                    return React.createElement(ContextMenuItem, {
                        key: item.className + '-' + index.text,
                        iconClassName: item.className,
                        text: item.text,
                        onClick: item.onClick
                    });
                })
            )
        )
    );
};

ContextMenu.defaultProps = {
    hide: true,
    onLayerClick: function onLayerClick() {},
    x: undefined,
    y: undefined,
    items: []
};

export default ContextMenu;
